Microsoft.Toolkit.Uwp/Helpers/SystemInformation.cs

```diff
@@ -121,6 +122,17 @@ public static async Task LaunchStoreForReviewAsync()
         /// </summary>
         public PackageVersion FirstVersionInstalled { get; }
 
+        /// <summary>
+        /// Gets the previous version of the app that was installed.
+        /// This will be the current version if a previous version of the app was installed
+        /// before using <see cref="SystemInformation"/> or if the app is not updated.
+        /// </summary>
+        public PackageVersion PreviousVersionInstalled
+        {
+            get => _previousVersionInstalled;
+            private set => _previousVersionInstalled = value;
+        }
```

Comment:
Not really a fan of this change (from https://github.com/windows-toolkit/WindowsCommunityToolkit/pull/4000/commits/c883eb9ec54d40f7a0c6e58bf85b53d80057ba3f), which was done after my last review. I think it'd be better to leave all these properties as readonly, to make the code clearer and also to reduce the chance of errors in the future. Could you refactor the code to keep the same logic as today, but with a readonly property like before?

Reply:
Can we use that pattern outside the constructor still though?

Reply:
We can't, and that's by design, we'd have to set that from the constructor like the other properties. I'd argue that'd actually make the code easier to read as well. For instance, we could change the code to something like:

```csharp
private (bool IsUpdated, PackageVersion PreviousVersion) DetectIfAppUpdated()
{
    var currentVersion = ApplicationVersion.ToFormattedString();

    if (!_localObjectStorageHelper.KeyExists(nameof(currentVersion)))
    {
        _localObjectStorageHelper.Save(nameof(currentVersion), currentVersion);
    }
    else
    {
        var previousVersion = _localObjectStorageHelper.Read<string>(nameof(currentVersion));
        if (currentVersion != lastVersion)
        {
            _localObjectStorageHelper.Save(nameof(currentVersion), currentVersion);
            return (true, previousVersion.ToPackageVersion());
        }
    }

    return (false, currentVersion.ToPackageVersion());
}
```

And then the constructor would just do:

```csharp
(IsAppUpdated, PreviousVersionInstalled) = DetectIfAppUpdated();
```

This way it'd be much easier while reading the constructor to immediately see where every property is assigned from 🙂
Thoughts?

Reply:
Yeah, this is an interesting idea, want to make it a suggest on the code itself?

Reply:
For future reference: applied this change in https://github.com/windows-toolkit/WindowsCommunityToolkit/pull/4000/commits/76cf98838469d1f531022fbc23637cbf073bbcc4, see below.